File: src/main/java/org/visual/provider/SceneProvider.java
package org.visual.provider;

import jakarta.inject.Provider;
import javafx.scene.Scene;
import javafx.scene.layout.Pane;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class SceneProvider implements Provider<Scene> {
  @Override
  public Scene get() {
    return new Scene(new Pane());
  }
}

File: src/main/kotlin/org/visual/provider/SceneProvider.kt
package org.visual.provider

import jakarta.inject.Inject
import jakarta.inject.Provider
import javafx.scene.Scene
import org.visual.component.MainLayout

class SceneProvider @Inject constructor(private val pane: MainLayout) : Provider<Scene> {
  override fun get(): Scene {
    return Scene(pane).apply {}
  }
}
